### DISCUSSION TITLE
Facebook provider frontend Redirect url

### DISCUSSION THREAD

**Discussion by negati-ve** [strapi/strapi#6533] [Legacy Topics]
I'm running on Strapi 3.0.0 beta 20

I have a personal site running on strapi with a react frontend
the frontend allows users to login and strapi's facebook provider sends the request back to the specified frontend redirect URL 

Now, I want to setup another site for testing, with the same reactjs frontend and same strapi backend but the new test site has different domain name

as you can guess.. the problem is when I try to login to the new site hosted on new domain, 
strapi's facebook provider, after authentication still redirects to the old domain.. 

is there a way I can tell strapi to redirect to the new domain instead of the one old one registered in the strapi admin ?

![image](https://user-images.githubusercontent.com/13253073/83650683-3e98b100-a5d6-11ea-9265-67bd61e2a573.png)

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question that is not really a bug report or a feature request, questions should be directed to the [discussions](https://github.com/strapi/strapi/discussions). Please see the following contributing guidelines for asking a question [here](https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md#reporting-an-issue).

I have converted this to a discussion for you, we try to keep Github clean and only for bug reports and feature requests.

**Reply by negati-ve** [reply]
Thank you for  converting this to a discussion.. 

I posted this in issues cause it  seems that passing ?callback=url  to  facebook provider (/connect/facebook/)  still  seems to redirect to the url registered in strapi admin

I wasn't sure if this is fixed  in v3. 
trying out v3 right now

**Reply by negati-ve** [reply]
@derrickmehaffy Please advice, 
shall I create a new issue?

**Comment by negati-ve** [comment]
Hey @derrickmehaffy 
Looks like a bug to me

# Observed behavior
1. front end makes a call to http://localhost:1337/connect/facebook?callback=http://newdomain.com
2. strapi picks up the callback url from the url
3. strapi redirects to facebook for authentication 
4. facebook  redirects back  to  strapi at /connect/facebook/callback
4. strapi authenticates and redirects to http://olddomain.com/callback?access_token=xyz 

# Expected behavior
Assuming that callback parameter provided  in http://localhost:1337/connect/facebook?callback=http://newdomain.com
is  the  callback  after strapi authentication is completed AKA frontend redirect URI
1. front end makes a call to http://localhost:1337/connect/facebook?callback=http://newdomain.com
2. strapi picks up the callback url from the url
3. strapi redirects to facebook for authentication 
4. facebook  redirects back  to  strapi at /connect/facebook/callback
4. strapi authenticates and redirects to http://newdomain.com/callback?access_token=xyz

**Reply by negati-ve** [reply]
I'm trying this  on strapi 3.0.1

**Comment by negati-ve** [comment]
suspecting an issue with grant.. 

created a new [PR](https://github.com/simov/grant/pull/169) at grant with potential fix

**Comment by negati-ve** [comment]
@simov from grant pointed out that I need to  add dynamic:["callback"] to my facebook config. that fixed it!
